lib/pos_epoch.go

```diff
@@ -0,0 +1,170 @@
+package lib
+
+import (
+	"bytes"
+	"github.com/dgraph-io/badger/v3"
+	"github.com/pkg/errors"
+)
+
+//
+// TYPE
+//
+
+type EpochEntry struct {
+	EpochNumber            uint64
+	LastBlockHeightInEpoch uint64
```

Comment:
Equally clear but a bit more concise
```suggestion
	FinalBlockHeight uint64
```